File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
         int length=0;
        ListNode temp=head;
        while(temp!=null){
            temp=temp.next;
            length++;
        }
        if(length==1){
            return null;
        }
        int num=length-n+1;
        ListNode prev=null;
        temp=head;
        while(num!=1){
            prev=temp;
            temp=temp.next;
            num--;
        }
        if(prev!=null){
             prev.next=temp.next;
        }else{
            head=head.next;
        }
       
       
        return head;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        if head is None:
            return head
        l=0
        temp=head
        while temp:
            temp=temp.next
            l+=1
        stop=l-n+1
        temp=head
        prev=None
        while stop!=1:
            prev=temp
            temp=temp.next
            stop-=1
        if prev is None:
            head=head.next
        else:
            prev.next=temp.next
        return head
        
        

        